### PR TITLE
 fixes #27 by allowing users to add a new FoodItem into Datastore

### LINKED ISSUE #27: Need a form to add new foods into our database
We currently have a form to log what you have eaten. We need another form that will allow users to upload information about other foods.

I think that the simplest way to do this is to:
  - Create a new form to collect this data
  - Make both the eaten food form and new food form include a hidden input field that says what form is being filled out.
  - update the [FormHandlerServlet here](https://github.com/Ivyzhang1998/team-16-codeu/blob/master/src/main/java/com/google/codeu/servlets/FormHandlerServlet.java) to dispatch to different form processing functions based on the hidden input field.

To elaborate a bit more on the steps above. Inside of the forms we'll need to add something like `<input type="hidden" name="formType" value="eatenFood">` and `<input type="hidden" name="formType" value="newFood">`.

Then in the servlet, the `doPost` method will need to be updated to be something like

```
public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
  String formType = request.getParameter("formType");
  if (formType.isEmpty()) {
    // Return some kind of error message to the user.
  }
  if (formType.equals("eatenFood")) {
    handleEatenFood(request, response);
  } else if (formType.equals("newFood")) {
    handleNewFood(request, response);
  } else {
    // Return some kind of error message to the user.
  }
}
```

### PR DESCRIPTION
**Pull request by dayhahm**
The user is now able to enter their own CO2 data into Datastore through a form. The form is then handled by the same form handler servlet as the eaten food form, but through a hidden input field, the servlet handles each form differently. Also adds the user id as part of the food meal. Addresses #27

### CODE REVIEW

File: src/main/java/com/google/codeu/data/EatenMeal.java [hunk 1]
```diff
@@ -2,22 +2,23 @@
...
+  private UUID id;
+  private String userId;
+  private String food;
+  private Double amount;
   private Date date; //Date(int year, int month, int date)
   private String imageUrl;
   private MealType mealType;
 
-  public EatenMeal (List<String> foods, List<Double> amounts, Date date , String imageUrl, MealType mealType) {
+  public EatenMeal (String userId, String foods, Double amount, Date date , String imageUrl, MealType mealType) {
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/data/EatenMeal.java]: `foods` --> `food`

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -56,8 +56,10 @@ public void storeFood(FoodItem food) {
   /** Stores Meal in Datastore. */
   public void storeMeal(EatenMeal meal) {
     Entity mealEntity = new Entity("UserMeal", meal.getID().toString());
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/data/Datastore.java]: Here we should probably update this Datastore entity name

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -56,8 +56,10 @@ public void storeFood(FoodItem food) {
...
-    mealEntity.setProperty("Foods", meal.getFoods());
-    mealEntity.setProperty("Amounts", meal.getAmounts());
+    mealEntity.setProperty("Food", meal.getFood());
+    mealEntity.setProperty("UserId", meal.getUserId());
+    mealEntity.setProperty("Amount", meal.getAmount());
+    mealEntity.setProperty("MealType", meal.getMealType());
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/data/Datastore.java]: Do you know how Datastore handles enums like this?

File: src/main/java/com/google/codeu/data/EatenMeal.java [hunk 4]
```diff
@@ -27,12 +28,14 @@ public UUID getID() {
     return id;
   }
 
-  public List<String> getFoods() {
-    return foods;
+  public String getUserId() { return userId; }
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/data/EatenMeal.java]: For sake of consistency, this should be 
```
public String getUserId() {
    return userId;
}
```

File: src/main/java/com/google/codeu/servlets/Co2FoodFormPageServlet.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/co2-food-form")
+public class Co2FoodFormPageServlet extends HttpServlet{
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        request.getRequestDispatcher("/WEB-INF/views/co2-food-form.jsp").forward(request, response);
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/servlets/Co2FoodFormPageServlet.java:14]: Great work with these request dispatchers!

**Review by NickChapman — COMMENTED** [review]
Couple of comments, but overall lookin' really good! I _am_ requesting changes but I'm just going to add these things as a comment so that you're not blocked on me taking another look.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 3 above)
Comment (dayhahm) [reply src/main/java/com/google/codeu/data/Datastore.java]: I did some quick googling and it seems like it's fine when storing, but when querying you need to declare parameters. It seems kind of complicated, so I think i'm going to use the ordinal method to assign integer values to the enums and store that int instead

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 6]
```diff
@@ -55,15 +55,14 @@ public void storeFood(FoodItem food) {
 
   /** Stores Meal in Datastore. */
   public void storeMeal(EatenMeal meal) {
-    Entity mealEntity = new Entity("UserMeal", meal.getID().toString());
+    Entity mealEntity = new Entity("EatenMeak", meal.getID().toString());
```
Comment (Ivyzhang1998) [inline src/main/java/com/google/codeu/data/Datastore.java]: Small typo -> Should be EatenMeal

**Review by Ivyzhang1998 — COMMENTED** [review]
Feel free to merge it once you fix  this type!

**Review by Ivyzhang1998 — APPROVED** [review]
please fix the typo! otherwise looks good

**Review by Open-Fuse — APPROVED** [review]

File: src/main/java/com/google/codeu/servlets/Co2FoodFormPageServlet.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    }
+}
```
Comment (Ivyzhang1998) [inline src/main/java/com/google/codeu/servlets/Co2FoodFormPageServlet.java:16]: new line

**Pull request merged**